### ISSUE TITLE
Enable checker framework on CI

### ISSUE THREAD

**Issue by DavyLandman** [usethesource/rascal-language-servers#359]
We've tried to use null annotations correctly, but we should start checking those with CF. And reduce cases of nulls leaking where we don't expect them to leak.

**Comment by jurgenvinju** [comment]
I'm guessing this will strand on the fact the the rascal project has not been annotated and we use a lot of that API. Shouldn't we wait until we only depend on the much smaller and newer runtime of the compiled Rascal code?

**Comment by DavyLandman** [comment]
True, that would be the only part where we'll get false negatives from. Both vallang and lsp4j have annotations, and quite some of the code is written in a null-safe way as well.

It's just that this code is already mostly behaving in a compatible way. But yeah, this is an enhancement that we don't need to pick up now, just something that we could.

**Comment by DavyLandman** [comment]
Solved in #901
